fms/models/qwen3.py

```diff
@@ -0,0 +1,701 @@
+ "architectures": [
+    "Qwen3ForCausalLM"
+  ],
+  "attention_bias": false,
+  + "attention_dropout": 0.0,           # p_dropout (05)
+  "bos_token_id": 151643,               # added 
+  "eos_token_id": 151645,
+  "head_dim": 128,
+  + "hidden_act": "silu",               # activation_fn (01)
+  + "hidden_size": 2048,                # emb_dim (02), hidden_grow_factor (11)
+  "initializer_range": 0.02,
+  + "intermediate_size": 6144,          # inner_dim (hidden_grow_factor/11)
+  + "max_position_embeddings": 40960,   # max_expected_seq_len (03)
+  "max_window_layers": 28,
+  "model_type": "qwen3",
+  + "num_attention_heads": 16,          # nheads (09)
+  + "num_hidden_layers": 28,            # nlayers (10)
+  + "num_key_value_heads": 8,           # kvheads (04)
+  + "rms_norm_eps": 1e-06,              # norm_eps (06)
+  "rope_scaling": null,
+  + "rope_theta": 1000000,              # rope_base (07)
+  "sliding_window": null,
+  + "tie_word_embeddings": true,        # tie_heads (12)
+  - "torch_dtype": "bfloat16",
+  - "transformers_version": "4.51.0",
+  "use_cache": true,
+  "use_sliding_window": false,
+  + "vocab_size": 151936                # src_vocab_size (08)
+"""
+
+
+@dataclass
+class Qwen3Config(ModelConfig):
+
+    # From above mapping
+    src_vocab_size: int = 151936
+    nheads: int = 16
+    nlayers: int = 28
+    hidden_grow_factor: float = 6144 / 2048  # intermediate_size / hidden_size:emb_dim
+    tie_heads: bool = True
+    p_dropout: float = 0.0
+    activation_fn: str = "silu"
+    emb_dim: int = 2048
+    max_expected_seq_len: int = 40960
+    kvheads: int = 8
+    norm_eps: float = 1e-6
+    rope_base: float = 100_0000.0  # Same as rope_theta
+    # Extra?
+    bos_token_id:int = 151643
+    multiple_of: int = 256  # borrowed from llama
+    sliding_window: int = 4000
+    fused_weights: bool = True  # FMS Specific -- For CPU/GPU = T, AIU = F
+    pad_id: int = -1  # borrowed from granite, we do need it
+    linear_config: Optional[Mapping[str, Any]] = None  # To support quantization
+    # From config.json just in case?
+    attention_bias: bool = False
+    attention_dropout: float = 0.0
+    # bos_token_id: int = 151643
+    eos_token_id: int = 151645
+    head_dim:int = 128
+    hidden_act:str = "silu"
+    hidden_size: int = 2048
+    initializer_range: float = 0.02
+    intermediate_size:int = 6144
+    max_position_embeddings:int = 40960
+    max_window_layers:int = 28
+    model_type:str = "qwen3"
+    num_attention_heads:int = 16
+    num_hidden_layers:int = 28
+    num_key_value_heads:int = 8
+    rms_norm_eps:float = 1e-06
+    rope_scaling =  None
+    rope_theta:int = 1000000
+    sliding_window = 0  # not sure what to put here
+    tie_word_embeddings:bool = True
+    torch_dtype: str = "bfloat16"
+    transformers_version:str = "4.51.0"
+    use_cache:bool = True
+    use_sliding_window = False
+    vocab_size: int = 15193
```

Comment:
you are repeating many parameters with different names, please carefully map the HF ones to FMS and not more, you should have at most max(FMS params, HF params), not a sum of both